msal-dotnet-articles/acquiring-tokens/web-apps-apis/on-behalf-of-flow.md

```diff
@@ -51,11 +51,53 @@ private void AddAccountToCacheFromJwt(IEnumerable<string> scopes, JwtSecurityTok
   var application = BuildConfidentialClientApplication(httpContext, principal);
 
   // .Result to make sure that the cache is filled-in before the controller tries to get access tokens
-  var result = await application.AcquireTokenOnBehalfOf(
-           requestedScopes.Except(scopesRequestedByMsalNet),
-           userAssertion)
-              .ExecuteAsync()                     
+  var result = await application.AcquireTokenOnBehalfOf(requestedScopes, userAssertion).ExecuteAsync()                     
+}
+```
+
+## Handling multi-factor auth (MFA), conditional access and incremental consent
+
+
+### Failure scenario
+
+It is a common scenario that a tenant administrator restricts access to the downstream API (for example to Graph), by requiring end-users to MFA. But often they do not place the same restrictions on the web api. 
+
+1. The client (e.g. desktop app or web site) asks for a token for your web api. MFA is not enforced at this point!
+2. The web api tries to exchange this token for a token for the downstream web api (e.g. Graph), through on-behalf-of. This fails, because access through Graph requires the user to have MFA-ed. The call to `AcquireTokenOnBehalfOf` will fail with an `MsalUiRequiredException` which will also have the `Claims` property set.
```

Comment:
```suggestion
2. The web API tries to exchange this token for a token for the downstream web API (e.g. Graph) via the on-behalf-of flow. This fails because access through Graph requires the user to have completed the MFA challenge. The call to `AcquireTokenOnBehalfOf` will fail with an `MsalUiRequiredException` which will also have the `Claims` property set.
```